torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -519,6 +522,24 @@ void LoopNest::initialize(
       }
     } else {
       loops.push_back(loop);
+      if (For* f = dynamic_cast<For*>(loop)) {
+        worklist.push_back(f);
+      }
+    }
+  }
+
+  // Traverse all inner-most loops to build index-loop mapping
+  while (worklist.size()) {
+    For* f = worklist.back();
+    index_loop_mapping[f->var()] = f;
```

Comment:
Constructing the index map here is not sufficient, since loop transformations will create/destroy loops and their corresponding index variables. If we do not keep the map up to date, the simplifier will eventually crash trying to access destroyed var/loop and will not be able to simplify new loops. Keeping the map up to date across all transformations is error-prone and laborious, so a better way would be to collect the same info in the IRSimplifier itself as it visits the statements - this way the info will always be up to date.